service/history/nDCBranchMgr.go

```diff
@@ -139,7 +141,7 @@ func (r *nDCBranchMgrImpl) createNewBranch(
 	defer func() {
 		if errComplete := r.historyV2Mgr.CompleteForkBranch(&persistence.CompleteForkBranchRequest{
 			BranchToken: newBranchToken,
-			Success:     retError == nil || persistence.IsTimeoutError(retError),
+			Success:     true, // past lessons learnt from Cassandra & gocql tells that we cannot possibly find all timeout errors
```

Comment:
What is it means?

Reply:
this means that basically for all new branch created, the logic will mark it as success.
this is due to lessons learnt from using Cassandra & gocql, e.g. a error a timeout, or '*only receive 1 / 2 response' error, or other errors, does not really means that the query issued to Cassandra will fail.

the only safe thing to do it mark the branch creation as success, and tolerate data leak, until our background cleanup logic is implemented